ConfAnalyser/cyclohexane.cpp

```diff
@@ -156,32 +167,75 @@ bool Cyclohexane::is_boat() const
         double right_dist = plane.distance_from(*(C[(begin+2)%6]));
         double left_dist = plane.distance_from(*(C[(begin+5)%6]));
         return (abs(right_dist) > tolerance_out &&
-                abs(left_dist) > tolerance_out) &&
-               (right_dist * left_dist > 0);
+                abs(left_dist) > tolerance_out) &&      // is the distance of atoms out of the plane far enough?
+               (right_dist * left_dist > 0);            // are the left and right planes distances in positive values to make the boat shape?
 }
 
+/**
+ * @brief Checks if the ring is of boat shape with right twist.
+ * 
+ * @return true 
+ * @return false 
+ */
+bool Cyclohexane::is_tw_boat_right() const
+{
+        /* twisted boat has no plane within the circle */
+        if (has_plane) {
+                return false;
+        }
+        Plane_3D right_plane(*(C[begin]),
+                             *(C[(begin+1)%6]),
+                             *(C[(begin+3)%6]));
+        Plane_3D left_plane(*(C[begin]),
+                            *(C[(begin+1)%6]),
+                            *(C[(begin+4)%6]));
+        double right_dist = right_plane.distance_from(*(C[(begin+2)%6]));
+        double left_dist = left_plane.distance_from(*(C[(begin+5)%6]));
+        double tw_angle = dihedral_angle(*(C[(begin+1)%6]), *(C[(begin+3)%6]),
+					 *(C[(begin+4)%6]), *(C[begin]));
+
+        return ((abs(tw_angle) > angle_tw_boat - angle_tolerance) &&
+                (abs(tw_angle) < angle_tw_boat + angle_tolerance)) &&   // is the twist angle in allowed range?
+               ((abs(right_dist) > tolerance_tw_out) &&
+                (abs(left_dist) > tolerance_tw_out)) &&                 // is the distance of atoms out of the plane far enough?
+                (right_dist * left_dist > 0) &&                         // are the left and right planes distances in positive values to make the boat shape?
+                (right_dist > left_dist);                               // is the right plane distance higher to make the right twist boat?
+}
 
-bool Cyclohexane::is_tw_boat() const
+/**
+ * @brief Checks if the ring is of boat shape with left twist.
+ * 
+ * @return true 
+ * @return false 
+ */
```

Comment:
should describe the principles an, I guess, this will be delivered in another merge request